src/main/java/Duke/Commands/CommandMark.java

```diff
@@ -0,0 +1,25 @@
+package Duke.Commands;
+
+import Duke.DukeException.DukeException;
+import Duke.System.Storage;
+import Duke.System.TaskList;
+import Duke.System.Ui;
+
+public class CommandMark extends Command{
+    private final int taskNum;
+
+    public CommandMark(String str) throws DukeException {
```

Comment:
As mentioned in the Command.java, you could call Command's constructor and use that here. This will allow you to run the execute function without any attributes. Again this is more of a design choice I would say,